azure-keyvault-cryptography/src/test/java/com/microsoft/azure/keyvault/cryptography/test/ECKeyTest.java

```diff
@@ -0,0 +1,208 @@
+    static MessageDigest DIGEST_256;
+    static MessageDigest DIGEST_384;
+    static MessageDigest DIGEST_512;
+    static KeyPairGenerator EC_KEY_GENERATOR;
+    static Map<JsonWebKeyCurveName, MessageDigest> CURVE_TO_DIGEST;
+    static List<JsonWebKeyCurveName> CURVE_LIST;
+       
+    @BeforeClass
+    public static void setUpBeforeClass() throws Exception {
+    	EC_KEY_GENERATOR = KeyPairGenerator.getInstance("EC", new BouncyCastleProvider());
+
+    	DIGEST_256 = MessageDigest.getInstance("SHA-256");
+    	DIGEST_384 = MessageDigest.getInstance("SHA-384");
+    	DIGEST_512 = MessageDigest.getInstance("SHA-512");
+    	
+    	CURVE_TO_DIGEST = ImmutableMap.<JsonWebKeyCurveName, MessageDigest>builder()
+			.put(JsonWebKeyCurveName.P_256, DIGEST_256)
+			.put(JsonWebKeyCurveName.P_384, DIGEST_384)
+			.put(JsonWebKeyCurveName.P_521, DIGEST_512)
+			.put(JsonWebKeyCurveName.SECP256K1, DIGEST_256)
+			.build();
+    	
+    	CURVE_LIST = Arrays.asList(JsonWebKeyCurveName.P_256, JsonWebKeyCurveName.P_384, JsonWebKeyCurveName.P_521, JsonWebKeyCurveName.SECP256K1);
+    }
+    
+    @Test
+    public void testCurves() throws Exception {
+    	for (JsonWebKeyCurveName crv : CURVE_LIST) {
+    		EcKey key = new EcKey("keyId", crv);
+    		doSignVerify(key, CURVE_TO_DIGEST.get(crv));
+    	}
+    }
+    
+    @Test(expected = NoSuchAlgorithmException.class)
+    public void testUnsupportedCurve() throws NoSuchAlgorithmException, InvalidAlgorithmParameterException, InstantiationException, IllegalAccessException, ClassNotFoundException {
+    	EcKey key = new EcKey("keyId", new JsonWebKeyCurveName("not an algo"));
+    }
+    
+    @Test
+    public void testDefaultKey() throws Exception {
+    	EcKey key = new EcKey("keyId");
+    	doSignVerify(key, DIGEST_256);
+    }
+    
+    @Test
+    public void testWithBCProvider() throws Exception {
+    	// BC provider is the default anyway.
+    	EcKey key = new EcKey("keyId", JsonWebKeyCurveName.P_256, new BouncyCastleProvider());
+    	doSignVerify(key, DIGEST_256);
+    }
+    
+    @Test
+    public void testWithKeyPair() throws Exception {
+    	for (JsonWebKeyCurveName crv : CURVE_LIST) {
+        	ECGenParameterSpec gps = new ECGenParameterSpec(EcKey.CURVE_TO_SPEC_NAME.get(crv));
+        	EC_KEY_GENERATOR.initialize(gps);
+        	KeyPair keyPair = EC_KEY_GENERATOR.generateKeyPair();
+        	
+        	final String name = "keyid";
+        	EcKey key = new EcKey(name, keyPair);
+        	assertEquals(name, key.getKid());
+        	assertEquals(crv, key.getCurve());
+        	doSignVerify(key, CURVE_TO_DIGEST.get(crv));
+    	}
+    }
+    
+    @Test(expected = NoSuchAlgorithmException.class)
+    public void testWithNotCurveKeyPair() throws Exception {
+    	ECGenParameterSpec gps = new ECGenParameterSpec("secp192k1");

```

Comment:
>ECGenParameterSpec gps = new ECGenParameterSpec("secp192k1"); [](start = 5, length = 61)

shouldn't this be testing SECP256K1. our service doesn't support secp192k1.

Reply:
Here I was testing a not-supported algorithm